test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2368,166 +1442,324 @@ describe('AwsCompileFunctions', () => {
+
+        expect(MemorySize).to.equal(providerConfig.memorySize);
+      });
+
+      it('should prefer `functions[].memorySize` over `provider.memorySize`', () => {
+        const fooFunctionConfig = serviceConfig.functions.foo;
+
+        const { MemorySize } = cfResources[naming.getLambdaLogicalId('foo')].Properties;
+
+        expect(MemorySize).to.equal(fooFunctionConfig.memorySize);
+      });
+
+      it('should support `provider.runtime`', () => {
+        const providerConfig = serviceConfig.provider;
+
+        const { Runtime } = cfResources[naming.getLambdaLogicalId('other')].Properties;
+
+        expect(Runtime).to.equal(providerConfig.runtime);
+      });
+
+      it('should prefer `functions[].runtime` over `provider.runtime`', () => {
+        const fooFunctionConfig = serviceConfig.functions.foo;
+
+        const { Runtime } = cfResources[naming.getLambdaLogicalId('foo')].Properties;
+
+        expect(Runtime).to.equal(fooFunctionConfig.runtime);
+      });
+
+      it('should support `provider.versionFunctions: false`', () => {
+        expect(cfOutputs).to.not.have.property(naming.getLambdaVersionOutputLogicalId('other'));
+      });
+
+      it('should prefer `functions[].versionFunction` over `provider.versionFunctions`', () => {
+        expect(cfOutputs).to.have.property(naming.getLambdaVersionOutputLogicalId('foo'));
+      });
+
+      it('should support `provider.deploymentBucket`', () => {
+        const providerConfig = serviceConfig.provider;
+
+        const {
+          Code: { S3Bucket },
+        } = cfResources[naming.getLambdaLogicalId('foo')].Properties;
+
+        expect(S3Bucket).to.deep.equal(providerConfig.deploymentBucket);
+      });
+
+      it('should support `functions[].fileSystemConfig` with vpc configured on provider', () => {
+        const functionServiceConfig = serviceConfig.functions.fnFileSystemConfig;
+        const fileSystemCfConfig =
+          cfResources[naming.getLambdaLogicalId('fnFileSystemConfig')].Properties;
+
+        const { arn, localMountPath } = functionServiceConfig.fileSystemConfig;
+        expect(arn).to.match(/^arn/);
+        expect(localMountPath).to.be.a('string');
+        expect(fileSystemCfConfig.FileSystemConfigs).to.deep.equal([
+          { Arn: arn, LocalMountPath: localMountPath },
+        ]);
+        expect(iamRolePolicyStatements).to.deep.include({
+          Effect: 'Allow',
+          Action: ['elasticfilesystem:ClientMount', 'elasticfilesystem:ClientWrite'],
+          Resource: [arn],
+        });
+      });
     });
 
-    it('should support `functions[].fileSystemConfig` with vpc configured on provider', () => {
-      const functionServiceConfig = serviceConfig.functions.fnFileSystemConfig;
-      const fileSystemCfConfig =
-        cfResources[naming.getLambdaLogicalId('fnFileSystemConfig')].Properties;
+    describe('when IamRoleLambdaExecution is not used', () => {
```

Comment:
Let's title it `when custom IAM role is used` (`IamRoleLambdaExecution` is an internal name, I think we should not attach to it in tests, also for users not that familiar with internals it's likely not be clear what `IamRoleLambdaExecution` is exactly)

Reply:
done, I've also adjusted the naming/variables a bit to avoid shadowing them from the higher scope